src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/AssessmentSummaryQueryTest.kt

```diff
@@ -23,22 +23,37 @@ class AssessmentSummaryQueryTest : IntegrationTestBase() {
 
   @Test
   fun `assessment summary query works as described when not restricted to user`() {
-    `Given a User` { user, _ ->
-      `Given an Assessment for Approved Premises`(user, user, reallocated = true) { _, _ -> }
-      `Given an Assessment for Approved Premises`(user, user) { apAssessment, _ ->
-        val expectedApAssessmentNote = earliestUnansweredClarificationNote(apAssessment, user)
-        `Given an Assessment for Temporary Accommodation`(user, user) { taAssessment, _ ->
-          val expectedTaAssessmentNote = earliestUnansweredClarificationNote(taAssessment, user)
-
-          val results: List<DomainAssessmentSummary> = realAssessmentRepository.findAllAssessmentSummariesNotReallocated()
-
-          assertThat(results.size).isEqualTo(2)
-
-          results.forEach {
-            when (it.id) {
-              apAssessment.id -> assertForAssessmentSummary(it, apAssessment, expectedApAssessmentNote.createdAt)
-              taAssessment.id -> assertForAssessmentSummary(it, taAssessment, expectedTaAssessmentNote.createdAt)
-              else -> fail()
+    `Given a User` { user1, _ ->
+      `Given an Assessment for Approved Premises`(user1, user1, reallocated = true) { _, _ -> }
+      `Given an Assessment for Approved Premises`(user1, user1) { apAssessment, _ ->
+
+        val expectedApAssessmentNote = earliestUnansweredClarificationNote(apAssessment, user1)
+        `Given an Assessment for Temporary Accommodation`(user1, user1) { taAssessment, _ ->
+          val expectedTaAssessmentNote = earliestUnansweredClarificationNote(taAssessment, user1)
+          `Given a User` { user2, _ ->
+
+            val u2Assessment = assessmentEntityFactory.produceAndPersist {
```

Comment:
Not for now, but I was talking to @brownbearsoftware the other day about adding cucumber-like comments to the integration tests to make them easier to follow. This one is quite small, and the Fixture stuff gives a lot of the `Given`-type context, but sometimes there can be a lot of code running through these tests and it can get quite tricky to grok

Reply:
Kotest BehaviourSpec might help, although the given/when/then blocks have to be nested.
https://kotest.io/docs/framework/testing-styles.html#behavior-spec

Kotest assertions are much more readable the junit / assertj assertions (in my opinion):
https://kotest.io/docs/assertions/assertions.html

Reply:
Yeah, too much nesting can make things much much worse! I was just thinking comments are a light touch solution. I like Kotest assertions though - much more like Rspec matchers, which, as a Rubyist, makes me very happy! Probably too late to switch out wholesale though.